### PR TITLE
docs: fix some typos and confusing wording

### PR DESCRIPTION
**Pull request by DyXel**
This is what I caught on my first reading. Looks good so far!

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -152,7 +152,7 @@ Here we can see more of how Cpp2 makes it features work.
 **How: Simple and safe by default.**
 
 - **Line 9:** CTAD just works, because it turns into ordinary C++ code which is CTAD-aware.
-- **Lines 10-11:** The accesses of `words[0]` and `words[1]` are bounds-checked nonintrusively at the call site by default. Because it's nonintrusive, it works seamlessly with all existing container types that are `std::ssize`-aware, when you use them from safe Cpp2 code.
```
[the remark this answers is not in the document]
Comment (DyXel) [reply docs/index.md:155]: I know, but if you look earlier in the document it does say `std::size`, so I changed this to be consistent with the earlier mention. Which version should we keep? `std::size`, `std::ssize`, both,  something else?

As additional context, what the doc say is "`std::size`-aware" really means satisfying this concept:
https://github.com/hsutter/cppfront/blob/main/include/cpp2util.h#L531-L532

[the remark this answers is not in the document]
Comment (gregmarr) [reply docs/index.md:155]: I've started a review, and updated this in my local version to list both of them.  For details, it uses `ssize` when you use a signed value and `size` for an `unsigned` value.

**Comment by DyXel** [comment]
@gregmarr feel free to steal all the changes in this PR and let me know so I can close it in favor of yours.

**Comment by gregmarr** [comment]
@DyXel probably easier for you to just remove that one.  It's going to be a while before I get much further through it. I'm traveling tomorrow.

**Comment by hsutter** [comment]
Thanks!

**Pull request merged**